AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyWidgetContainer.py

```diff
@@ -0,0 +1,52 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects used when testing and manipulating the O3DE UI
+"""
+import pyside_utils
+import editor_python_test_tools.QtPyScriptCanvasEditor as QtPyScriptCanvasEditor
+import editor_python_test_tools.QtPyScriptCanvasVariableManager as QtPyScriptCanvasVariableManager
+
+
+class QtPyWidgetContainer:
+    """
+    Container class for QtPy objects used in testing/manipulating the UI
+
+    """
+
+    def __init__(self):
+        self.editor_main_window = pyside_utils.get_editor_main_window()
+        self.sc_editor = None
+        self.variable_manager = None
+
+    def initialize_SC_objects(self):
+        """
+        Function for instantiating the two core script canvas QtPy objects used in SC testing.
+        sc_editor is the editor itself, which the other objects are anchored or embedded in
+        sc_editor_main_window is the main pane that houses the graph UI and basic controls
+
+        returns None
+        """
+        self.sc_editor = QtPyScriptCanvasEditor.QtPyScriptCanvasEditor(self.editor_main_window)
```

Comment:
Why leave it to the user to initialize this, and not initialize inside `__init__`?

Reply:
Related to the question about holding state: Why wouldn't a user simply instantiate their own QtPyScriptCanvasEditor directly?

Reply:
We want the usage of this container to be a bit more modular. By grouping the Qt objects like the initialize_SC_objects function we can reduce the number of lines used to set up say a script canvas test and avoid slowing down other people's tests if they aren't testing around script canvas.